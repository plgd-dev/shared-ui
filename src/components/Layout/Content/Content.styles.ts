import { css } from '@emotion/react'
import { ThemeType, get } from '../../Atomic/_theme'

export const content = (theme: ThemeType) => css`
    background: ${get(theme, `Content.background`)};
    padding: 24px 40px 0 24px;
`

export const header = css`
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding-bottom: 16px;
`

export const left = css`
    display: flex;
    align-items: center;
    flex: 1 1 auto;
`

export const rightActions = css`
    display: flex;
    padding: 0;
    margin: 0 -4px;
    list-style: none;
`

export const rightAction = css`
    display: flex;
    padding: 0 4px;
`

export const headline = (theme: ThemeType) => css`
    color: ${get(theme, `Content.headline.color`)};
`

export const statusTag = css`
    padding-left: 8px;
`
