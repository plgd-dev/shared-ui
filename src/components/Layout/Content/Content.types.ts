import { ReactNode } from 'react'
import { ButtonVariantsType } from '../../Atomic/Button/Button.types'

export type ContentActionType = {
    icon: string
    text: string
    variant?: ButtonVariantsType
    onClick: () => void
}

export type Props = {
    actionPrimary?: ContentActionType
    actions?: ContentActionType[]
    children?: ReactNode
    headline: string
    headlineStatusTag?: ReactNode
}
