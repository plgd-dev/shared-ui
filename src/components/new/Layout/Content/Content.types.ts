import { ReactNode } from 'react'

export type Props = {
    headline: string
    headlineStatusTag?: ReactNode
    actions: ReactNode[]
}
