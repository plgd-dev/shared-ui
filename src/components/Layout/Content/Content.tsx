import { FC, memo, useMemo } from 'react'
import { Props } from './Content.types'
import * as styles from './Content.styles'
import Headline from '../../Atomic/Headline'
import Button from '../../Atomic/Button'
import { Icon } from '../../Atomic/Icon'

type ContentHeaderProps = {
    headerProps: Omit<Props, 'children'>
}

const ContentHeader: FC<ContentHeaderProps> = memo((props) => {
    const { headline, headlineStatusTag, actions } = props.headerProps
    return (
        <div css={styles.header}>
            <div css={styles.left}>
                <Headline css={styles.headline} type='h4'>
                    {headline}
                </Headline>
                {headlineStatusTag && <div css={styles.statusTag}>{headlineStatusTag}</div>}
            </div>
            {actions && (
                <ul css={styles.rightActions}>
                    {actions.map((action, key) => (
                        <li css={styles.rightAction} key={key}>
                            <Button icon={<Icon icon={action.icon} />} onClick={action.onClick} variant={action.variant}>
                                {action.text}
                            </Button>
                        </li>
                    ))}
                </ul>
            )}
        </div>
    )
})

const Content: FC<Props> = (props) => {
    const { actions, headline, headlineStatusTag, children } = props
    const headerProps = useMemo(() => ({ actions, headline, headlineStatusTag }), [actions, headline, headlineStatusTag])

    return (
        <div css={styles.content}>
            <ContentHeader headerProps={headerProps} />
            {children}
        </div>
    )
}

Content.displayName = 'Content'

export default Content
