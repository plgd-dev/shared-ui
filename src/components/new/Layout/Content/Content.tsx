import { FC } from 'react'
import { Props } from './Content.types'
import * as styles from './Content.styles'
import Headline from '../../Headline'

const Content: FC<Props> = (props) => {
    const { actions, headline } = props
    return (
        <div css={styles.content}>
            <div css={styles.header}>
                <div css={styles.left}>
                    <Headline css={styles.headline} type='h4'>
                        {headline}
                    </Headline>
                </div>
                {actions && (
                    <ul css={styles.rightActions}>
                        {actions.map((action, key) => (
                            <li css={styles.rightAction} key={key}>
                                {action}
                            </li>
                        ))}
                    </ul>
                )}
            </div>
        </div>
    )
}

Content.displayName = 'Content'

export default Content
