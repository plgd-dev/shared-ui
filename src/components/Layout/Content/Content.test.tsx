import { render } from '../../../../test/jest-wrapper'
import Content from './Content'

describe('Content Component', () => {
    it('renders headline correctly', () => {
        const { getByText, asFragment } = render(<Content headline='Test Headline' />)
        expect(getByText('Test Headline')).toBeInTheDocument()
        expect(asFragment()).toMatchSnapshot()
    })

    it('renders headline status tag when provided', () => {
        const { getByText } = render(<Content headline='Test Headline' headlineStatusTag='Test Status' />)
        expect(getByText('Test Status')).toBeInTheDocument()
    })

    it('renders children correctly', () => {
        const { getByText } = render(
            <Content headline='Test Headline'>
                <div>Test Child</div>
            </Content>
        )
        expect(getByText('Test Child')).toBeInTheDocument()
    })
})
