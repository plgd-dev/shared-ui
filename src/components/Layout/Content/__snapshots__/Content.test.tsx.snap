// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content Component renders headline correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  background: #FFFFFF;
  padding: 24px 40px 0 24px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding-bottom: 16px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
}

.emotion-3 {
  color: #0A2965;
  font-family: 'Circular Pro',sans-serif;
  font-weight: 700;
  font-style: normal;
  font-size: 24px;
  line-height: 30px;
  letter-spacing: -0.5px;
  margin: 0;
  color: #0A2965;
}

@media (max-width: 767px) {
  .emotion-3 {
    font-size: 20px;
    line-height: 24px;
  }
}

<div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <h4
          class="h4 emotion-3"
        >
          Test Headline
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
